lib/Backend/GlobOpt.cpp

```diff
@@ -4877,6 +4877,40 @@ GlobOpt::OptInstr(IR::Instr *&instr, bool* isInstrRemoved)
         this->KillStateForGeneratorYield();
     }
 
+    // Change LdFld on arrays, strings, and 'arguments' to LdLen when we're accessing the .length field
+    if (instr->GetSrc1() && instr->GetSrc1()->GetKind() == IR::OpndKindSym && instr->m_opcode == Js::OpCode::ProfiledLdFld || instr->m_opcode == Js::OpCode::LdFld || instr->m_opcode == Js::OpCode::ScopedLdFld)
+    {
+        IR::Opnd * opnd = instr->GetSrc1();
+        Sym *sym = opnd->AsSymOpnd()->m_sym;
+        if (sym->IsPropertySym()) {
+            PropertySym *originalPropertySym = sym->AsPropertySym();
+            IR::RegOpnd* newopnd = IR::RegOpnd::New(originalPropertySym->m_stackSym, IRType::TyVar, instr->m_func);
+            // only on .length
+            if (this->lengthEquivBv != nullptr && this->lengthEquivBv->Test(originalPropertySym->m_id)) {
+                Value *const objectValue = FindValue(originalPropertySym->m_stackSym);
+                // Only for things we'd emit a fast path for
+                if (
+                    objectValue->GetValueInfo()->IsLikelyAnyArray() ||
+                    objectValue->GetValueInfo()->HasHadStringTag() ||
+                    objectValue->GetValueInfo()->IsLikelyString() ||
+                    newopnd->IsArgumentsObject() ||
```

Comment:
>IsArgumentsObject [](start = 29, length = 17)

The IsArgumentsOpnd() check below is a more precise check for arguments object. You could get rid of this IsArgumentsObject check, unless you kept it for short circuiting